resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DataprocWorkerChecker.scala

```diff
@@ -0,0 +1,67 @@
+package com.broadinstitute.dsp.resourceValidator
+
+import cats.effect.{Concurrent, Timer}
+import cats.mtl.Ask
+import cats.implicits._
+import com.broadinstitute.dsp.{
+  regionName,
+  resourceValidator,
+  CheckRunner,
+  CheckRunnerConfigs,
+  CheckRunnerDeps,
+  RuntimeCheckerDeps,
+  RuntimeWithWorkers
+}
+import io.chrisdavenport.log4cats.Logger
+import org.broadinstitute.dsde.workbench.google2.DataprocClusterName
+import org.broadinstitute.dsde.workbench.model.TraceId
+
+object DataprocWorkerChecker {
+  def impl[F[_]: Timer](
+    dbReader: DbReader[F],
+    deps: RuntimeCheckerDeps[F]
+  )(implicit F: Concurrent[F], logger: Logger[F], ev: Ask[F, TraceId]): CheckRunner[F, RuntimeWithWorkers] =
+    new CheckRunner[F, RuntimeWithWorkers] {
+      override def appName: String = resourceValidator.appName
+
+      override def configs = CheckRunnerConfigs(s"dataproc-workers", shouldAlert = true)
+
+      override def dependencies: CheckRunnerDeps[F] = deps.checkRunnerDeps
+
+      override def resourceToScan: fs2.Stream[F, RuntimeWithWorkers] = dbReader.getRuntimesWithWorkers
+
+      override def checkResource(runtime: RuntimeWithWorkers, isDryRun: Boolean)(
+        implicit ev: Ask[F, TraceId]
+      ): F[Option[RuntimeWithWorkers]] =
+        for {
+          clusterOpt <- deps.dataprocService
+            .getCluster(runtime.r.googleProject, regionName, DataprocClusterName(runtime.r.runtimeName))
+          runtime <- clusterOpt.fold[F[Option[RuntimeWithWorkers]]](F.pure(None)) { c =>
+            val doesPrimaryWorkerMatch =
+              runtime.workerConfig.numberOfWorkers == c.getConfig.getWorkerConfig.getNumInstances
+            val doesSecondaryWorkerMatch =
+              runtime.workerConfig.numberOfPreemptibleWorkers == c.getConfig.getSecondaryWorkerConfig.getNumInstances
+            val isAnomalyDetected = !(doesPrimaryWorkerMatch && doesSecondaryWorkerMatch)
+
+            isAnomalyDetected match {
+              case true =>
+                val log =
+                  if (isDryRun)
+                    logger
+                      .warn(
+                        s"${runtime} has an anomaly with the number of workers in google. \n\tPrimary work match status: ${doesPrimaryWorkerMatch}\n\tSecondary worker match status: ${doesSecondaryWorkerMatch}"
```

Comment:
primary worker?

Reply:
it can be both, thats why i log and record both

Reply:
I meant you have a typo `Primary work match status`